### ISSUE TITLE
Body write doesn't wait for async finish?

### ISSUE THREAD

**Issue by wclr** [cprussin/purescript-httpure#175]
If I get it right this Aff actually doesn't wait for the finish of writing and ending the stream, is it intentional?
```purs
  write body response =
    Aff.makeAff \done -> do
      let
        stream = HTTP.responseAsStream response
      _ <- Stream.writeString stream Encoding.UTF8 body $ pure unit
      _ <- Stream.end stream $ pure unit
      done $ Either.Right unit
      pure Aff.nonCanceler
```

**Comment by cprussin** [comment]
Hmm that's a good point.  I'm not sure it really matters due to the way that's threaded into the underlying Node http server, although for sure it's probably semantically incorrect.  Is it causing any specific problems or did you just see that the code looked strange?

I'd totally be happy to accept a PR to fix it if you wanted to submit one!

**Comment by wclr** [comment]
I don't think that it might cause an issue with the current workflow, the node internally handles the stuff so you may call `end` before the last `write` is finished it will end it up first. So If there is no async error/result handling assumed this will work out fine. 

> did you just see that the code looked strange?

That's right.

**Comment by cprussin** [comment]
@wclr Fixed in #176 .  Let me know if you see any urgent need to cut a version with the fix, and thanks for pointing it out!

**Comment by wclr** [comment]
Also, there is a problem with `Node.Stream` write api, that it doesn't allow to handle errors: https://github.com/purescript-node/purescript-node-streams/issues/35

This would be really needed for the proper implementation of Server-Sent Events.
